data-prepper-plugins/kafka-source/src/main/java/org/opensearch/dataprepper/plugins/kafka/source/consumer/PlainTextConsumer.java

```diff
@@ -0,0 +1,114 @@
+  private static final Logger LOG = LoggerFactory.getLogger(PlainTextConsumer.class);
+  private Map<TopicPartition, OffsetAndMetadata> offsetsToCommit = new HashMap<>();
+  private long lastReadOffset = 0L;
+  private KafkaConsumer<String, String> plainTxtConsumer;
+  private volatile long lastCommitTime = System.currentTimeMillis();
+  private static final Long COMMIT_OFFSET_INTERVAL_MILLISEC = 300000L;
+
+  @SuppressWarnings({ "unchecked", "rawtypes" })
+  @Override
+  public void consumeRecords(KafkaConsumer<String, String> consumer, AtomicBoolean status,
+      Buffer<Record<Object>> buffer, TopicConfig topicConfig, PluginMetrics pluginMetrics) {
+    KafkaSourceBufferAccumulator kafkaSourceBufferAccumulator = new KafkaSourceBufferAccumulator(topicConfig, pluginMetrics);
+    plainTxtConsumer = consumer;
+    try {
+      consumer.subscribe(Arrays.asList(topicConfig.getName()));
+      while (!status.get()) {
+        ConsumerRecords<String, String> records = consumer.poll(Duration.ofMillis(100));
+        if (!records.isEmpty() && records.count() > 0) {
+          for (TopicPartition partition : records.partitions()) {
+            List<Record<Object>> kafkaRecords = new ArrayList<>();
+            List<ConsumerRecord<String, String>> partitionRecords = records.records(partition);
+            for (ConsumerRecord<String, String> consumerRecord : partitionRecords) {
+              offsetsToCommit.put(new TopicPartition(consumerRecord.topic(), consumerRecord.partition()),
+                 new OffsetAndMetadata(consumerRecord.offset() + 1, null));
+              kafkaRecords.add(kafkaSourceBufferAccumulator.getEventRecord(consumerRecord.value(),topicConfig));
+              lastReadOffset = partitionRecords.get(partitionRecords.size() - 1).offset();
+            }
+            if (!kafkaRecords.isEmpty()){
+              kafkaSourceBufferAccumulator.writeAllRecordToBuffer(kafkaRecords,buffer,topicConfig);
+            }
+          }
+          if(!offsetsToCommit.isEmpty()) {
+            commitOffsets(consumer);
+          }
+        }
+      }
+    } catch (Exception exp) {
+      LOG.error("Error while reading plain text records from the topic...{}", exp.getMessage());
+    }
+  }
+
+  @Override
+  public void onPartitionsAssigned(Collection<TopicPartition> partitions) {
+    LOG.trace("onPartitionsAssigned() callback triggered and Closing the consumer...");
+    for (TopicPartition partition : partitions) {
+      plainTxtConsumer.seek(partition, lastReadOffset);
+    }
+  }
+
+  @Override
+  public void onPartitionsRevoked(Collection<TopicPartition> partitions) {
+    LOG.trace("onPartitionsRevoked() callback triggered and Committing the offsets: {} ", offsetsToCommit);
+    try {
+      plainTxtConsumer.commitSync(offsetsToCommit);
+    } catch (CommitFailedException e) {
+      LOG.error("Failed to commit the record...", e);
+    }
+  }
+
+  public void commitOffsets(KafkaConsumer<String, String> consumer) {
+    try {
```

Comment:
Also, we should structure it such a way that `commitOffsets` is in one place instead of in multiple places like PlainTextConsumer.java and JsonConsumer.java etc.